### ISSUE TITLE
Anvil eth_estimateGas fails with blob transaction

### ISSUE THREAD

**Issue by Thegaram** [foundry-rs/foundry#7843]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (d495216 2024-05-03T00:21:54.426103000Z)

### What command(s) is the bug in?

anvil

### Operating System

macOS (Apple Silicon)

### Describe the bug

I try to estimate gas before sending a blob transaction, however this fails with the following error:

```
"There should be at least one blob in a Blob transaction."
```

This seems to be inconsistent with geth, where it's not possible to provide blobs for `eth_call`, and `eth_estimateGas` through to Go SDK, only `BlobHashes`, which I do provide in this case. https://github.com/ethereum/go-ethereum/blob/master/interfaces.go#L158

**Comment by Thegaram** [comment]
Steps to reproduce:

```shell
> anvil --chain-id 111111 --hardfork cancun
```

```shell
> curl -H "Content-Type: application/json" -X POST --data '{
	"jsonrpc": "2.0",
	"method": "eth_estimateGas",
	"params": [{
		"from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
		"to":"0x9Af51261a1f324Df094dc8028F2E5259C23B5336",
		"data": "0xffffffff",
		"maxFeePerGas": "0x1",
		"maxPriorityFeePerGas": "0x1",
		"blobVersionedHashes": ["0x5cf6f95ae1c6694e63fb6c6a4395f609c60d71496aa7d7a146c63be45cfd21b6"],
		"maxFeePerBlobGas": "0x1"
	}],
	"id": 1
}' -s http://127.0.0.1:8545 | jq

{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "code": -32003,
    "message": "There should be at least one blob in a Blob transaction."
  }
}
```

**Comment by Thegaram** [comment]
It seems that geth generally disallows providing a blob for these APIs: https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/transaction_args.go#L299

**Comment by mattsse** [comment]
interesting, checking

thanks for flagging

**Comment by mattsse** [comment]
@yash-atreya yeah for call+estimate we shouldn't require a blob because not useful

**Comment by yash-atreya** [comment]
@mattsse Fix for this was included here 181dc3c5b704b1e6284ed82509b150b31bd77d61 and it was released in the latest nightly.
